quodlibet/quodlibet/ext/events/seekpoints.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# For more information, please refer to <http://unlicense.org>
+
+from gi.repository import Gtk
+from quodlibet import _, app
+from quodlibet.plugins import PluginConfigMixin
+from quodlibet.plugins.songshelpers import has_bookmark
+from quodlibet.plugins.events import EventPlugin
+from quodlibet.qltk import Icons
+from quodlibet.qltk.entry import UndoEntry
+from quodlibet.qltk.tracker import TimeTracker
+
+
+class SeekPointsPlugin(EventPlugin, PluginConfigMixin):
+    """The plugin class."""
+
+    PLUGIN_ID = "Seekpoints"
+    PLUGIN_NAME = _("Seekpoint Bookmarks")
+    PLUGIN_ICON = Icons.GO_JUMP
+    PLUGIN_CONFIG_SECTION = __name__
+    PLUGIN_DESC = _(
+       "Store Seekpoints A and/or B for tracks. "
+       "Skip to time A and stop after time B when track is played.\n"
+       "Note that changing the names of the points below does not "
+       "update the actual bookmark names, it only changes which "
+       "bookmark names the plugin looks for when deciding whether to seek.")
+
+    CFG_SEEKPOINT_A_TEXT = "A"
+    CFG_SEEKPOINT_B_TEXT = "B"
+    DEFAULT_A_TEXT = "A"
+    DEFAULT_B_TEXT = "B"
+
+    def enabled(self):
+        self._SeekPoint_A, self._SeekPoint_B = self._get_SeekPoints()
+        self._try_create_tracker(force=True)
+
+    def disabled(self):
+        self._try_destroy_tracker()
+
+    def _try_create_tracker(self, force=False):
+        """Create the tracker if it does not exist,
+           or if forced to, like at plugin enable"""
+        if force or not self._tracker_is_enabled:
+            self._tracker = TimeTracker(app.player)
+            self._tracker.connect('tick', self._on_tick)
+            self._tracker_is_enabled = True
+
+    def _try_destroy_tracker(self):
+        """Destroy the tracker if it exists"""
+        if self._tracker_is_enabled:
+            self._tracker.destroy()
+            self._tracker_is_enabled = False
+
+    # Seeks to point A if it exists
+    def plugin_on_song_started(self, song):
+        self._try_create_tracker()
+        self._SeekPoint_A, self._SeekPoint_B = self._get_SeekPoints()
+        if not self._SeekPoint_A:
+            return
+        self._seek(self._SeekPoint_A)
+
+    # Finishes track after point B has been reached, if it exists
+    def _on_tick(self, tracker):
+        if not self._SeekPoint_B:
+            self._try_destroy_tracker()
+            return
+
+        time = app.player.get_position() // 1000
+        if self._SeekPoint_B <= time:
+            self._seek(app.player.info("~#length"))
+
+    def _get_SeekPoints(self):
```

Comment:
Please use lowercase function names and variable names. https://www.python.org/dev/peps/pep-0008/#function-names